### PR TITLE
Docs: Miscellaneous 

### PR DESCRIPTION
**Pull request by noraabiakar**
* `concepts/cable_cell`: number subcategories to better illustrate document structure.
* `concepts/labels`: concretize -> thingify; more on label dictionaries.
* `concepts/mechanisms`: Some info on modcc, link to nmodl files in repo.
* `concepts/morphology`: NeuroML section and some notes on branches/segments

### CODE REVIEW

File: doc/concepts/labels.rst [hunk 1]
```diff
@@ -1,7 +1,7 @@
 .. _labels:
 
-Cell labels
-===========
+Labels
```
Comment (brenthuisman) [inline doc/concepts/labels.rst:3]: With the earlier reordering of all the cable_cell specific pages at the bottom of the hierarchy, removing 'Cell' from these page titles will make it not obvious that these are Cell (and in fact cable_cell) specific. I initially added them as middle ground between creating another layer in the page hierarchy and having no association in the hierarchy at all.

If you really wish to remove 'Cell', what's a good way to group these subcomponents of the cable cell?

File: doc/concepts/labels.rst [hunk 2]
```diff
@@ -559,20 +559,21 @@ Region expressions
...
-Concretisation
-----------------
+Thingification
+--------------
 
-When a region or locset expression is applied to a cell morphology it is
-*concretised*. Concretising a locset will return a set of *locations* on the
-morphology, and concretising a region will return a list of unbranched *cables*
-on the morphology.
+When a region or locset expression is applied to a cell morphology, it is represented
```
Comment (brenthuisman) [inline doc/concepts/labels.rst:567]: Yes! Now I know what thingification actually is :)

File: doc/concepts/mechanisms.rst [hunk 3]
```diff
@@ -1,16 +1,23 @@
 .. _mechanisms:
 
-Cell mechanisms
-===============
+Mechanisms
```
Comment (brenthuisman) [inline doc/concepts/mechanisms.rst:3]: See grouping comment.

File: doc/concepts/morphology.rst [hunk 4]
```diff
@@ -1,15 +1,16 @@
 .. _morph:
 
-Cell morphology
-===============
+Morphology
```
Comment (brenthuisman) [inline doc/concepts/morphology.rst:3]: See grouping comment.

**Review by brenthuisman — CHANGES_REQUESTED** [review]

File: doc/concepts/labels.rst (hunk 1 above)
Comment (noraabiakar) [reply doc/concepts/labels.rst:3]: I'll make it `Cable cell labels`

**Review by brenthuisman — APPROVED** [review]
gtg!

**Pull request merged**